internal/authentication/ldap_user_provider.go

```diff
@@ -197,29 +148,30 @@ type ldapUserProfile struct {
 	Username    string
 }
 
-func (p *LDAPUserProvider) resolveUsersFilter(userFilter string, inputUsername string) string {
-	inputUsername = p.ldapEscape(inputUsername)
+func (p *LDAPUserProvider) resolveUsersFilter(inputUsername string) (filter string) {
+	filter = p.configuration.UsersFilter
 
-	// The {input} placeholder is replaced by the users username input.
-	userFilter = strings.ReplaceAll(userFilter, "{input}", inputUsername)
+	if p.usersFilterReplacementInput {
+		// The {input} placeholder is replaced by the users username input.
+		filter = strings.ReplaceAll(filter, "{input}", p.ldapEscape(inputUsername))
+	}
 
-	p.logger.Tracef("Computed user filter is %s", userFilter)
+	if p.usersFilterReplacementEpochWin32 {
+		filter = strings.ReplaceAll(filter, "{epoch:win32}", strconv.FormatUint(utils.Win32EpochNow(), 10))
+	}
```

Comment:
gobench indicates this adds at most 1ns to this function, but potentially saves 100-150ns per check for instances where the replacement doesn't exist in the filter.